frontend_app_file/src/component/FileComponent.jsx

```diff
@@ -7,36 +7,36 @@ import PreviewComponent from './PreviewComponent.jsx'
 import { MODE } from '../helper.js'
 import FileDropzone from './FileDropzone.jsx'
 import PopupProgressUpload from './PopupProgressUpload.jsx'
+import { DisplayState } from 'tracim_frontend_lib'
 
 export const FileComponent = props =>
   <div className={classnames('file__contentpage__statewrapper', {'displayState': props.isArchived || props.isDeleted})}>
-    {props.isArchived &&
-      <div className='file__contentpage__preview__state'>
-        <div className='file__contentpage__preview__state__msg'>
-          <i className='fa fa-fw fa-archive' />
-          {props.t('This content is archived')}
-        </div>
-
-        <button className='file__contentpage__preview__state__btnrestore btn' onClick={props.onClickRestoreArchived}>
-          <i className='fa fa-fw fa-archive' />
-          {props.t('Restore')}
-        </button>
-      </div>
-    }
+    {props.isArchived && (
+      <DisplayState
+        msg={props.t('This content is archived')}
+        btnType='button'
+        icon='archive'
+        btnLabel={props.t('Restore')}
+        onClickBtn={props.onClickRestoreArchived}
+      />
+    )}
 
-    {props.isDeleted &&
-      <div className='file__contentpage__preview__state'>
-        <div className='file__contentpage__preview__state__msg'>
-          <i className='fa fa-fw fa-trash' />
-          {props.t('This content is deleted')}
-        </div>
+    {props.isDeleted && (
+      <DisplayState
+        msg={props.t('This content is deleted')}
+        btnType='button'
+        icon='trash'
+        btnLabel={props.t('Restore')}
+        onClickBtn={props.onClickRestoreDeleted}
+      />
+    )}
 
-        <button className='file__contentpage__preview__state__btnrestore btn' onClick={props.onClickRestoreDeleted}>
-          <i className='fa fa-fw fa-trash' />
-          {props.t('Restore')}
-        </button>
-      </div>
-    }
+    {props.isDeprecated && (
+      <DisplayState
+        msg={props.t('This content is deprecated')}
+        icon='warning'
```

Comment:
The icon should come from state.config.availableStatuses.
Use the improper way `state.config.availableStatuses[3]` for now will fix it later.

Reply:
This component does not have access to the state. Should I pass the icon as a props?

Reply:
yes you have to
Only give the deprecated status for now.
If, one day FileComponent needs another one, we'll refactor the props to give the full availableStatuses array